examples/per_platform_printer/BUILD

```diff
@@ -0,0 +1,36 @@
+package(default_visibility = ["//visibility:public"])
+
+load(
+    "@io_bazel_rules_rust//rust:rust.bzl",
+    "rust_binary",
+    "rust_library",
+    "rust_test",
+)
+
+rust_binary(
+    name = "print",
+    srcs = ["main.rs"],
+    deps = [
+        ":printer",
+    ],
+)
+
+rust_library(
+    name = "printer",
+    srcs = [
+        "lib.rs",
+        "print_generic.rs",
+    ] + select({
+        "@io_bazel_rules_rust//rust/platform:x86_64-unknown-linux-gnu": [
```

Comment:
It feels a little weird to select on the triple when we only care about the os.

Are the individual constraints exposed at all? I had raised https://github.com/bazelbuild/bazel/issues/4167 at some point..

Reply:
Sorry, I wrote this with the context of the cargo-raze change in my head (which shall use full triples for ease of implementation). You're right, and golang also did the same thing where they generated config_settings at each level of granuarity.

https://github.com/bazelbuild/rules_go/blob/00adf001c27a5ef82c90e740e1d63509ead69ec4/go/platform/list.bzl#L86-L95

I've added them here as well now.